pre-commit

```diff
@@ -0,0 +1,9 @@
+# Get the paths of the staged .js, .ts, .jsx, .tsx files inside ./src
+files=$(git diff --cached --name-only --diff-filter=ACM | grep -E '^(./src/)?.*\.(ts|js|jsx|tsx)$')
```

Comment:
<div>

<strong>Bito Code Review Agent Run</strong> <a href=https://github.com/warmwei818/OpenAI-Code-Review-Demo/pull/3#issuecomment-2175135648>#2b2d36</a><span> - 06/18/2024, 06:19 am</span>
<br>

<div><small>🔴</small> High importance</div>

<div id="suggestion">
<div id="issue">
<b>Issue:</b> The command 'git diff --cached --name-only --diff-filter=ACM | grep -E '^(./src/)?.*\.(ts|js|jsx|tsx)$'' may not correctly handle file paths with spaces or special characters, leading to potential issues in identifying the correct files for review.
</div>
<div id="fix">
<b>Fix:</b> Use 'git diff --cached --name-only --diff-filter=ACM -z' to handle file paths with spaces and special characters, and adjust the 'grep' command accordingly.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -2,1 +2,1 @@
 -files=$(git diff --cached --name-only --diff-filter=ACM | grep -E '^(./src/)?.*\.(ts|js|jsx|tsx)$')
 +files=$(git diff --cached --name-only --diff-filter=ACM -z | grep -z -E '^(./src/)?.*\.(ts|js|jsx|tsx)$')
```
</div>
</details>
</div>
<br>


</div>